### PR TITLE
Add `IsSensitive` to `AttributeSchema`

### PR DESCRIPTION
**Pull request by beandrad**
So that sensitive attributes can be flagged.

### CODE REVIEW

File: decoder/attribute_candidates.go [hunk 1]
```diff
@@ -35,6 +35,10 @@ func detailForAttribute(attr *schema.AttributeSchema) string {
 		detail = "Optional"
 	}
 
+	if attr.IsSensitive {
```
Comment (beandrad) [inline decoder/attribute_candidates.go:38]: not sure about this...

**Review by radeksimko — COMMENTED** [review]
Thanks for the PR!

We'll need to reflect this new field also in hover data: https://github.com/hashicorp/hcl-lang/blob/b62f3a9965765599ec3bb707d628467b6afdd4a7/decoder/attribute_candidates.go#L30

Also would you mind adding some test cases for the public `CandidatesAtPos` here
https://github.com/hashicorp/hcl-lang/blob/b62f3a9965765599ec3bb707d628467b6afdd4a7/decoder/candidates_test.go#L803-L807

and similar for `HoverAtPos` here
https://github.com/hashicorp/hcl-lang/blob/b62f3a9965765599ec3bb707d628467b6afdd4a7/decoder/hover_test.go#L328-L332

One case in each (with sensitive attribute) should suffice; if you want to be thorough you could add one with `AttributeSchema` where `Expr` is `ObjectExpr` with sensitive attribute nested within, although I'm not sure what's the current handling of it as nested expressions aren't well supported yet - but it's still nice to have a test that reflects the current behaviour - and will fail (and be fixed) as part of implementing https://github.com/hashicorp/terraform-ls/issues/496

File: decoder/attribute_candidates.go (hunk 1 above)
Comment (radeksimko) [reply decoder/attribute_candidates.go:38]: Also for variable schema specifically we don't actually know (yet) whether a variable is required or optional without parsing the `default` - which we decided not to parse initially, so we should most certainly account for this eventuality here.

**Comment by radeksimko** [comment]
As per conversation in https://github.com/hashicorp/terraform-schema/pull/49#issuecomment-852166335 we should also add some tests which test `AttributeSchema` with no explicit required/optional during completion or hover. I think the validation should keep reporting it as failure/error, but we can treat it as "soft enforcement" for pragmatic reasons, so the decoder should account for this edge case and test for it.

File: decoder/attribute_candidates.go (hunk 1 above)
Comment (radeksimko) [reply decoder/attribute_candidates.go:38]: You could probably create a string slice, such as `details []string` and then `strings.Join()` everything together upon return 🤔  and I wouldn't worry too much about the capital letter in `sensitive` - it's an edge case anyway and it's kinda debatable whether optional/required should be capitalized anyway in this context 🤷🏻

File: decoder/attribute_candidates.go [hunk 2]
```diff
@@ -28,23 +28,24 @@ func attributeSchemaToCandidate(name string, attr *schema.AttributeSchema, rng h
 }
 
 func detailForAttribute(attr *schema.AttributeSchema) string {
-	var detail string
+	details := []string{}
+
 	if attr.IsRequired {
-		detail = "Required"
+		details = append(details, "required")
```
Comment (radeksimko) [inline decoder/attribute_candidates.go:34]: This is a reasonable change, but just FYI there will be some existing test data in the need of update potentially in `terraform-schema` and in `terraform-ls` 😅

**Review by radeksimko — APPROVED** [review]

File: decoder/attribute_candidates.go (hunk 2 above)
Comment (beandrad) [reply decoder/attribute_candidates.go:34]: sounds exciting  🙈
